src/coffea/nanoevents/mapping/uproot.py

```diff
@@ -48,6 +49,21 @@ def _lazify_form(form, prefix, docstr=None):
         )
         if parameters:
             form["parameters"] = parameters
+    elif form["class"] == "IndexedOptionArray":
+        if (
+            form["content"]["class"] != "NumpyArray"
+            or form["content"]["primitive"] != "bool"
+        ):
+            raise ValueError(
+                "Only boolean NumpyArrays can be created dynamically if "
+                "missing in file!"
+            )
+        form["form_key"] = quote(prefix + "allowmissing,!index")
```

Comment:
I assume `assert prefix.endswith("!load")` is true here?
Is there a way to instead use the DSL to itself construct the dummy array? I guess not in general because the form shouldn't know if a given chunk is or is not missing a field.

Reply:
Yeah the DSL doesn't know about the form key so you have to do it in the mapping.

Reply:
Maybe worth adding the assert for sake of later maintenance

Reply:
Will do, makes sense.